src/include/daos_srv/daos_server.h

```diff
@@ -623,6 +624,39 @@ d_rank_t dss_self_rank(void);
 
 unsigned int dss_ctx_nr_get(void);
 
+/* Cache for container root */
+struct tree_cache_root {
+	struct btr_root	btr_root;
+	daos_handle_t	root_hdl;
+	unsigned int	count;
+};
+
+int
+obj_tree_insert(daos_handle_t toh, uuid_t co_uuid, daos_unit_oid_t oid,
+		d_iov_t *val_iov);
+int
+obj_tree_destroy(daos_handle_t btr_hdl);
+
+/* Per xstream migrate status */
+struct ds_migrate_status {
+	uint64_t dm_rec_count;	/* migrated record size */
+	uint64_t dm_obj_count;	/* migrated object count */
+	uint64_t dm_total_size;	/* migrated total size */
+	int	 dm_status;	/* migrate status */
+	uint32_t dm_migrating:1; /* if it is migrating */
+};
+
+int
+ds_migrate_query_status(uuid_t pool_uuid, uint32_t ver,
+			struct ds_migrate_status *dms);
+int
+ds_object_migrate(struct ds_pool *pool, uuid_t pool_hdl_uuid, uuid_t cont_uuid,
+		  uuid_t cont_hdl_uuid, int tgt_id, uint32_t version,
+		  uint64_t max_eph, daos_unit_oid_t *oids, daos_epoch_t *ephs,
+		  unsigned int *shards, int cnt);
+void
+ds_migrate_fini_one(uuid_t pool_uuid, uint32_t ver);
+
```

Comment:
Perhaps we'd have some comments to explain what's a migration job? Can a migration job have multiple ds_object_migrate() calls? And I don't quite see how we distinguish different migration jobs on server side (by pool uuid + pool version? If so, it means there can be only one migration job for a pool in certain version? Do we need to pin pool_tls on server if a migration job have multiple ds_object_migrate() calls?).

This ds_migrate_fini_one() looks like used for aborting migration?

Reply:
Yes, used for both abort and finish.